plenum/server/replica.py

```diff
@@ -511,11 +329,61 @@ def __init__(self, node: 'plenum.server.node.Node', instId: int,
 
         HookManager.__init__(self, ReplicaHooks.get_all_vals())
 
-        self._consensus_data = ConsensusSharedData(self.node.name,
-                                                   self.node.get_validators(),
-                                                   self.instId)
+        self._bootstrap_consensus_data()
+        self._subscribe_to_internal_msgs()
         self._consensus_data_helper = ConsensusDataHelper(self._consensus_data)
 
+        self._external_bus = ExternalBus(send_handler=self.send)
+
+        self._ordering_service = OrderingService(self._consensus_data,
+                                                 timer=self.node,
+                                                 bus=self.node.internal_bus,
+                                                 network=self._external_bus,
+                                                 write_manager=self.node.write_manager,
+                                                 bls_bft_replica=self._bls_bft_replica,
+                                                 is_master=isMaster,
+                                                 get_current_time=self.get_current_time,
+                                                 get_time_for_3pc_batch=self.get_time_for_3pc_batch)
+        self.inBoxRouter = Router(
```

Comment:
For a legacy purpose, we can change `threePhaseRouter` subscription, so that legacy stashing logic works, but `threePhaseRouter` calls OrderingService's 3PC process methods.